eva_sub_cli/validators/validator.py

```diff
@@ -558,17 +558,38 @@ def _collect_file_info_to_metadata(self):
                                            file_name_2_md5.get(file_dict.get('fileName')) or ''
                         file_dict['fileSize'] = file_path_2_file_size.get(file_path) or \
                                                 file_name_2_file_size.get(file_dict.get('fileName')) or ''
+
+                        if not file_dict.get('fileSize'):
+                            error_txt = f"File size is not available for {file_dict.get('fileName')}"
+                            self.error(error_txt)
+                            errors.append({'property': '/files/fileSize', 'description': error_txt})
+                        if not file_dict.get('md5'):
+                            error_txt = f"md5 is not available for {file_dict.get('fileName')}"
+                            self.error(error_txt)
+                            errors.append({'property': '/files/md5', 'description': error_txt})
                         file_rows.append(file_dict)
                 else:
-                    self.error('No file found in metadata and multiple analysis alias exist: '
-                               'cannot infer the relationship between files and analysis alias')
+                    error_txt = ('No file section found in metadata and multiple analysis alias exist: '
+                                 'cannot infer the relationship between files and analysis alias')
+                    self.error(error_txt)
+                    errors.append({'property': '/files', 'description': error_txt})
                 metadata.set_files(file_rows)
             except Exception as e:
+                raise e
                 # Skip adding the md5
-                self.error('Error while loading or parsing metadata json: ' + str(e))
+                error_txt = 'Error while loading or parsing metadata json: ' + str(e)
+                self.error(error_txt)
+                errors.append({'property': '/', 'description': error_txt})
             metadata.write(self.metadata_json_post_validation)
         else:
-            self.error(f'Cannot locate the metadata in JSON format in {os.path.join(self.output_dir, "metadata.json")}')
+            error_txt = f'Cannot locate the metadata in JSON format in {os.path.join(self.output_dir, "metadata.json")}'
+            self.error(error_txt)
+            errors.append({'property': '/', 'description': error_txt})
```

Comment:
It's extra weird to have a JSON error which states that we can't find the JSON file, especially if the user didn't provide a JSON file and we never said we would create a JSON file... Clearly the process should fail at some point in this case, but I'm not convinced it belongs in the validation report.

Reply:
We should raise an exception instead.